File: pfe-java/app/controllers/Service.java
package controllers;

import com.google.inject.Singleton;
import models.AuctionRooms;
import models.Shop;
import models.SocialNetwork;
import models.Users;
import play.libs.ws.WSClient;
import play.libs.ws.WS;

@Singleton
public class Service {

    public final WSClient ws = WS.client();

    public final Shop shop = new Shop();

    public final AuctionRooms auctionRooms = new AuctionRooms();

    public final Users users = new Users();

    public final SocialNetwork socialNetwork = new SocialNetwork(ws);

}

File: pfe-scala/app/controllers/Service.scala
package controllers

import javax.inject.{Inject, Singleton}

import db.Schema
import models.{AuctionRooms, Shop, SocialNetwork, Users}
import play.api.libs.ws.WSClient

@Singleton class Service @Inject() (val app: play.api.Application) {

  val shop = new Shop(new Schema(app))

  val auctionRooms = new AuctionRooms(app)

  val users = new Users

  val socialNetwork = new SocialNetwork(app.injector.instanceOf[WSClient])

}
